packages/language-server/src/plugins/typescript/diagnostic-enhancements.ts

```diff
@@ -0,0 +1,65 @@
+import { CodeMapping } from "@volar/language-core";
+import type { Diagnostic } from "@volar/language-server";
+import { TextDocument } from "vscode-languageserver-textdocument";
+
+export function enhanceDiagnosticPositions(
```

Comment:
This is necessary to ensure we show diagnostics. Volar doesn't seem to display them when they are larger than the code mappings, so I clamp the diagnostics to the most reasonable code mapping we can find... 

There is almost certainly a better way, but this is good enough most of the time _I think_.